modules/ecs-service/main.tf

```diff
@@ -0,0 +1,692 @@
+    # be created and the module should fail.
+    postcondition {
+      condition     = self.status == "ACTIVE"
+      error_message = "The ECS cluster must be in an ACTIVE state before creating the ECS service. Currently the ECS cluster is ${self.status}."
+    }
+
+    # If the ECS service is using Service Connect, a default namespace must be
+    # configured for the ECS cluster.
+    postcondition {
+      condition     = !var.enable_service_connect || self.service_connect_defaults[0].namespace != null
+      error_message = "An ECS service using Service Connect must have a default namespace configured for the ECS cluster."
+    }
+  }
+}
+
+
+# -------------------------------------------
+
+# IDENTIFY THE TASK DEFINITION REVISION TO USE 
+
+# FOR THE ECS SERVICE OR SCHEDULED TASK
+
+# -------------------------------------------
+
+locals {
+  # This allows us to query both the existing as well as Terraform's state and get
+  # and get the max version of either source, useful for when external resources
+  # update the container definition
+  max_task_def_revision = max(aws_ecs_task_definition.task.revision, data.aws_ecs_task_definition.task.revision)
+  task_definition       = "${aws_ecs_task_definition.task.arn_without_revision}:${local.max_task_def_revision}"
+}
+
+# This allows us to query both the existing as well as Terraform's state and get
+# and get the max version of either source, useful for when external resources
+# update the container definition
+data "aws_ecs_task_definition" "task" {
+  task_definition = aws_ecs_task_definition.task.family
+
+  depends_on = [
+    aws_ecs_task_definition.task
+  ]
+}
+
+
+# ------------------------------------------------------------
+
+# THE FOLLOWING SECTION IS USED TO CONFIGURE THE ECS
+
+# TASK DEFINITION AND CONTAINER DEFINITION
+
+# ------------------------------------------------------------
+
+
+# -------------------------------------------
+# IDENTIFY IMAGE TO USE FOR THE TASK DEFINITION
+# -------------------------------------------
+
+locals {
+  # If the container image is not specified, then the latest version of the
+  # container definition image that is deployed will be used.
+  lookup_deployed_image = var.ecs_container_image == ""
+
+  # When looking up the deployed image, we need to distinguish between
+  # a scheduled task and a service. A scheduled task will not have a
+  # service associated with it
+  image = (
+    local.lookup_deployed_image ?
+    (
+      var.create_scheduled_task ?
+      data.aws_ecs_container_definition.scheduled[0].image :
+      data.aws_ecs_container_definition.service[0].image
+    )
+    : var.ecs_container_image
+  )
+}
+
+# -------------------------------------------
+# LOOKUP IMAGE FOR ECS SERVICE
+# -------------------------------------------
+
+# This will get the ECS service that is currently deployed.
+data "aws_ecs_service" "service" {
+  count = !var.create_scheduled_task && local.lookup_deployed_image ? 1 : 0
```

Comment:
Cant you make this a local variable? I notice you use the same count 3 times.

Reply:
For all of theses conditions in the whole file should I move all the different conditions to a locals block at the top or have the locals block with the section it corresponds to?